xposed-bridge/src/main/java/de/robv/android/xposed/XSharedPreferences.java

```diff
@@ -480,25 +489,39 @@ public Editor edit() {
         throw new UnsupportedOperationException("read-only implementation");
     }
 
-    @Deprecated
+    /**
+     * Registers a callback to be invoked when a change happens to a preference file.<br>
+     * Note that it is not possible to determine which preference changed exactly and thus
+     * preference key in callback invocation will always be null.
+     *
+     * @param listener The callback that will run.
+     * @see #unregisterOnSharedPreferenceChangeListener
+     */
     @Override
     public void registerOnSharedPreferenceChangeListener(OnSharedPreferenceChangeListener listener) {
-        if (!mWatcherEnabled)
-            throw new UnsupportedOperationException("File watcher feature is disabled for this instance");
-
         synchronized(this) {
-            mListeners.put(listener, sContent);
+            if (listener != null && !mListeners.containsKey(listener)) {
```

Comment:
If listener is null throw an exception.
Replace containsKey with null check on return value of put().